src/settings/SettingsArea.tsx

```diff
@@ -61,68 +64,195 @@ const SettingsArea = () => {
+              <NumberIncrementStepper />
+              <NumberDecrementStepper />
+            </NumberInputStepper>
+          </NumberInput>
+        </FormControl>
+      </VStack>
+      <VStack alignItems="flex-start" padding={3} spacing={5}>
+        <Text as="h2" fontWeight="semibold">
+          Code structure highlighting (experimental)
+        </Text>
+        <SelectFormControl
+          id="codeStructureShape"
+          label="Shape"
+          options={[
+            {
+              value: "l-shape",
+              label: "L-shapes",
+            },
+            {
+              value: "box",
+              label: "Boxes",
+            },
+          ]}
+          value={settings.codeStructureHighlighting.shape}
+          onChange={(shape) =>
+            setSettings(modifyCodeStructureSettings(settings, { shape }))
+          }
+        />
+        <SelectFormControl
+          id="codeStructureBorder"
+          label="Borders"
+          options={[
+            {
+              value: "none",
+              label: "None",
+            },
+            {
+              value: "borders",
+              label: "Borders",
+            },
+            {
+              value: "left-edge-only",
+              label: "Left edge only",
+            },
+          ]}
+          value={settings.codeStructureHighlighting.borders}
+          onChange={(borders) =>
+            setSettings(modifyCodeStructureSettings(settings, { borders }))
+          }
+        />
+        <BooleanFormControl
+          id="codeStructureHoverBackground"
+          label="Highlight background on mouse over"
+          value={settings.codeStructureHighlighting.hoverBackground || false}
+          onChange={(hoverBackground) =>
+            setSettings(
+              modifyCodeStructureSettings(settings, { hoverBackground })
+            )
+          }
+        />
+        <BooleanFormControl
+          id="codeStructureCursorBackground"
+          label="Highlight background for cursor position"
+          value={settings.codeStructureHighlighting.cursorBackground || false}
+          onChange={(cursorBackground) =>
+            setSettings(
+              modifyCodeStructureSettings(settings, { cursorBackground })
+            )
+          }
+        />
+
+        <BooleanFormControl
+          id="codeStructureCursorBorder"
+          label="Highlight border for cursor position"
+          value={settings.codeStructureHighlighting.cursorBorder || false}
+          onChange={(cursorBorder) =>
+            setSettings(modifyCodeStructureSettings(settings, { cursorBorder }))
+          }
+        />
+
+        <BooleanFormControl
+          id="codeStructureHoverBorder"
+          label="Highlight border on mouse over"
+          value={settings.codeStructureHighlighting.hoverBorder || false}
+          onChange={(hoverBorder) =>
+            setSettings(modifyCodeStructureSettings(settings, { hoverBorder }))
+          }
+        />
+      </VStack>
```

Comment:
These settings aren't implemented yet. I'll do that on Monday. 

Should just be adding few classes and faffing with the CSS for hover.

Cursor position will require us to add/update a class.